### PR TITLE
Crystal fix tasks of members are hidden when teams toggle is off

### PR DESCRIPTION
**Pull request by crystallow1168**
# Description
image

Bug priority high

## Related PRS (if any):
To test this backend PR you need to checkout the development frontend PR.

## Main changes explained:
- Implement condition so that users (besides Owner/Admin/Core Team) cannot view team's tasks when their team toggle visibility is off

## How to test:
1. Check into current branch:  `git checkout crystal-followup-fix-so-tasks-of-members-are-hidden-with-the-teams-toggle`
2. Do `npm install`, `npm run build` and `npm run start` to run this PR locally
3. Clear site data/cache
4. Log as Admin user
5. Go to Dashboard → Other Links→ Teams →…
6. Add members to the team to test. Verify that user with team visibility toggle off:
  - Owner role -> Can view other members tasks
User with Owner role visibility toggle off still can view other member’s task

  - Admin role -> Can view other members tasks
User with Admin role visibility toggle off can still view other member’s task

  - Core Team role -> Can view other members tasks
User with Core Team role visibility toggle off still can view other member’s task

  - Manager role -> CANNOT view other members tasks
User with Manager role visibility toggle off cannot view other member’s task

  - Volunteer role -> CANNOT view other members tasks
User with Volunteer role visibility toggle off cannot view other member’s task

  - Volunteer role (user has 2 teams)-> CANNOT view other members tasks 
User with Volunteer role visibility toggle off cannot view other member’s task, testing with 2 teams

## Screenshots or videos of changes:
How to set team member visibility's toggle. (refer to video below)

https://github.com/user-attachments/assets/16770b2e-a7a7-429c-9046-b51347128d01

Example of changes when testing user: Crystal Volunteer with Volunteer role.  (refer to video below)

https://github.com/user-attachments/assets/67a0e531-034b-42c3-8e77-9eebf630196a

### CODE REVIEW

File: src/helpers/taskHelper.js [hunk 1]
```diff
@@ -112,9 +112,15 @@ const taskHelper = function () {
           );
 
           sharedTeamsResult.forEach((_myTeam) => {
+            let hasTeamVisibility = false;
             _myTeam.members.forEach((teamMember) => {
-              if (!teamMember.userId.equals(userid)) teamMemberIds.push(teamMember.userId);
+              if (teamMember.userId.equals(userid) && teamMember.visible) hasTeamVisibility = true;
```
Comment (Ankuriboh) [inline src/helpers/taskHelper.js:117]: As long as the user that is requesting is the last user and was set as visible in settings, they could see all other users as well as their tasks. You didn't check if the user has necessary permission (i.e. if they're an Owner/Administrator/Manager).

File: src/helpers/taskHelper.js [hunk 2]
```diff
@@ -112,9 +112,15 @@ const taskHelper = function () {
...
             });
+            if (hasTeamVisibility) {
+              _myTeam.members.forEach((teamMember) => {
+                if (!teamMember.userId.equals(userid)) teamMemberIds.push(teamMember.userId);
+              });
+            }
```
Comment (Ankuriboh) [inline src/helpers/taskHelper.js:123]: Just `teamMembersIds.push(_myTeam.members.filter(m => !m.userId.equals(userid)).map(m => m.userId));`

**Review by Ankuriboh — CHANGES_REQUESTED** [review]
The code doesn't work on my end. It will always show other members' task whenever I've set disable my own visibility.

**Comment by crystallow1168** [comment]
> The code doesn't work on my end. It will always show other members' task whenever I've set to disable my own visibility.

Hello, @Ankuriboh did you set the visibility specifically from the Team page? 
Dashboard → Other Links→ Teams → Select the team which has the user you are testing -> turn off the toggle for the user that you are testing. I added a video to show how to set team member's visibility toggle.

https://github.com/user-attachments/assets/f00a616d-f14a-4a63-8569-1b3c57a0a94a

**Review by one-community — APPROVED** [review]

**Comment by Ankuriboh** [comment]
> > The code doesn't work on my end. It will always show other members' task whenever I've set to disable my own visibility.
> 
> Hello, @Ankuriboh did you set the visibility specifically from the Team page? Dashboard → Other Links→ Teams → Select the team which has the user you are testing -> turn off the toggle for the user that you are testing. I added a video to show how to set team member's visibility toggle.
> 
>  How.to.set.team.member.visibility.s.toggle.mp4

Hi, thanks for the kindly reply. It could probably be some problems on my end. But I tried on [frontend PR #2766 development preview](https://deploy-preview-2766--highestgoodnetwork-dev.netlify.app/dashboard), toggling off the "see all" option as an administrator wouldn't allow me others on the tasks page. Is this working as intended? Because the "see all" tips says that it does "NOT reduce visibility for Owner, Admin or Core Team roles".

**Pull request merged**